var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -65,8 +71,10 @@ def url_for_version(self, version):
     depends_on('latte', when='+latte')
     depends_on('blas', when='+latte')
     depends_on('lapack', when='+latte')
+    depends_on('python', when='+python')
 
     conflicts('+latte', when='@:20170921')
+    conflicts('+python', when='~lib')
```

Comment:
Can you make sure you have the dependencies of packages in here, too:
 https://github.com/lammps/lammps/blob/master/cmake/CMakeLists.txt#L123

Plus the `MPIIO` doesn't work without `MPI`.